torchtune/modules/peft/lora.py

```diff
@@ -128,6 +134,17 @@ def forward(self, x: Tensor) -> Tensor:
             return out
         lora_out = self.lora_a(self.dropout(x))
         lora_out = (self.alpha / self.rank) * self.lora_b(lora_out)
+        # Author mentions this method is faster for the computation purpose:
+        # https://github.com/huggingface/peft/pull/1474#issuecomment-1963402710
+        if self.use_dora:
+            weight_norm = torch.linalg.norm(
+                self.weight
+                + (self.alpha / self.rank)
+                * (self.lora_a.weight.T @ self.lora_b.weight.T).T,
+                dim=1,
+            ).to(self.weight.dtype)
```

Comment:
What about the detach used in the PEFT implementation referenced in your comment?